01_financial-well-being/housing-value.qmd

```diff
@@ -0,0 +1,904 @@
+
+```{r}
+#| label: keep-households-only
+acs1yr <-
+  acs1yr |>
+  filter(!(gq %in% c("Group quarters--Institutions", "Other group quarters")))
+```
+
+FIPS codes should always contain leading zeros so that state codes are two digits, county codes are three digits, and place codes are five digits.
+
+```{r}
+#| label: clean-fips-codes
+
+acs1yr <-
+  acs1yr |>
+  mutate(
+    across(c(state, county, city, puma), as.character),
+    state = str_pad(state, width = 2, pad = "0", side = "left"),
+    county = str_pad(county, width = 3, pad = "0", side = "left"),
+    city = str_pad(city, width = 5, pad = "0", side = "left"),
+    puma = str_pad(puma, width = 5, pad = "0", side = "left")
+  )
+```
+
+We create two age subgroups: households over 45 and households under 45
+
+```{r}
+#| label: add-age-subgroup
+
+acs1yr <-
+  acs1yr |>
+  mutate(subgroup = if_else(age >= 45, "Over or exactly 45 years old", "Under 45 years old"))
+```
+
+We collapse IPUMS race and ethnicity categories into white non-hispanic, Black non-hispanic, other non-hispanic, and
+
+```{r}
+#| label: add-race-ethnicity-subgroup
+
+
+acs1yr <-
+  acs1yr |>
+  mutate(
+    race = lbl_relabel(race, lbl(10, "Other") ~ .lbl %in% c("American Indian or Alaska Native", "Chinese", "Japanese", "Other Asian or Pacific Islander", "Other race, nec", "Two major races", "Three or more major races")),
+    across(c(race, hispan), as_factor)
+  ) |>
+  mutate(
+    race_ethnicity = case_when(
+      race == "White" & hispan == "Not Hispanic" ~ "white_nh",
+      race == "Black/African American" & hispan == "Not Hispanic" ~ "black_nh",
+      race == "Other" & hispan == "Not Hispanic" ~ "other_nh",
+      hispan != "Not Hispanic" ~ "hispanic",
+      .default = NA_character_
+    ),
+    race_ethnicity = factor(race_ethnicity,
+      levels = c(
+        "white_nh",
+        "black_nh",
+        "hispanic",
+        "other_nh"
+      )
+    )
+  )
+```
+
+## Merge Crosswalk {#sec-merge-crosswalk}
+
+We merge our 1-year ACS data with PUMA-county and PUMA-place geocodes to get county and place boundaries.
+
+For 2022 ACS and onward we use a crosswalk going from 2020 PUMA to 2020 county. For pre-2022 ACS data, we use a crosswalk from 2012 PUMA to 2014 county.
+
+The weight of each household is adjusted by the area of the PUMA that falls into a given County/Place
+
+```{r}
+#| label: define-crosswalk-years
+
+acs1yr <-
+  acs1yr |>
+  mutate(crosswalk_period = if_else(year < 2022, "pre-2022", "2022"))
+```
+
+::: panel-tabset
+#### County
+
+```{r}
+#| label: puma-to-county-crosswalk
+
+puma_county_crosswalk <-
+  read_csv(here::here(
+    "geographic-crosswalks",
+    "data",
+    "crosswalk_puma_to_county.csv"
+  )) |>
+  rename(
+    state = statefip,
+  ) |>
+  filter(state != "72")
```

Comment:
I'd mention you're filtering out PR in the text introducing this code.

Reply:
Added a comment about this! I'm not clear as to why we do this, but PR isn't present in the ACS data I pull from IPUMS so this filtering is done to avoid NA's later on.

Reply:
Thank you for adding this note!